File: demo/src/main/java/com/midas/demo/mvp/presenter/HotTabPresenter.java
package com.midas.demo.mvp.presenter;

import com.midas.demo.base.BasePresenter;
import com.midas.demo.mvp.contract.HotTabContract;
import com.midas.demo.mvp.model.HotTabModel;
import com.midas.demo.mvp.model.bean.TabInfoBean;
import io.reactivex.disposables.Disposable;
import io.reactivex.functions.Consumer;

public class HotTabPresenter extends BasePresenter<HotTabContract.View> implements HotTabContract.presenter {

    private HotTabModel model = new HotTabModel();

    @Override
    public void getTabInfo() {
        checkViewAttached();
        mViewRef.get().showLoading();

        Disposable disposable = model.getTabInfo().subscribe(new Consumer<TabInfoBean>() {
            @Override
            public void accept(TabInfoBean tabInfoBean) throws Exception {
                mViewRef.get().setTabInfo(tabInfoBean);
            }
        }, new Consumer<Throwable>() {
            @Override
            public void accept(Throwable throwable) throws Exception {
                mViewRef.get().showError("error");

            }
        });
        mViewRef.get().hideLoading();

        addSubscription(disposable);
    }
}

File: app/src/main/java/com/midas/eye/mvp/presenter/HotTabPresenter.kt
package com.midas.eye.mvp.presenter

import com.midas.eye.base.BasePresenter
import com.midas.eye.mvp.contract.HotTabContract
import com.midas.eye.mvp.model.HotTabModel
import com.midas.eye.net.exception.ExceptionHandle

/**
 * Created by midas on 2019/05/28.
 * desc: 获取 TabInfo Presenter
 */
class HotTabPresenter: BasePresenter<HotTabContract.View>(),HotTabContract.Presenter {

    private val hotTabModel by lazy { HotTabModel() }

    override fun getTabInfo() {
        checkViewAttached()
        mRootView?.showLoading()
        val disposable = hotTabModel.getTabInfo()
                .subscribe({
                    tabInfo->
                    mRootView?.setTabInfo(tabInfo)
                },{
                    throwable->
                    //处理异常
                    mRootView?.showError(ExceptionHandle.handleException(throwable), ExceptionHandle.errorCode)
                })
        addSubscription(disposable)
    }
}